systemtest/infra_kafka.go

```diff
@@ -21,233 +21,203 @@ import (
 	}
-	logger().Infof("provisioning Kafka infrastructure with config: %+v", cfg)
-	tf, err := NewTerraform(ctx, cfg.TFPath)
-	if err != nil {
-		return err
-	}
-	if err := tf.Init(ctx, tfexec.Upgrade(true)); err != nil {
-		return fmt.Errorf("failed to run terraform init: %w", err)
-	}
-	// Create a dummy topic so the infrastructure awaits until all the required
-	// infrastructure is available. It will be destroyed during the first test.
-	if len(cfg.Topics) == 0 {
-		cfg.Topics = append(cfg.Topics, SuffixTopics("dummy")...)
+
+	if v := os.Getenv("KAFKA_NAME"); v != "" {
+		kafkaClusterName = v
 	}
-	jsonTopics, err := json.Marshal(cfg.Topics)
-	if err != nil {
-		return fmt.Errorf("failed to marshal topics: %w", err)
+	if v := os.Getenv("KAFKA_NAMESPACE"); v != "" {
+		kafkaNamespace = v
 	}
-	namespaceVar := tfexec.Var(fmt.Sprintf("namespace=%s", cfg.Namespace))
-	nameVar := tfexec.Var(fmt.Sprintf("name=%s", cfg.Name))
-	topicsVar := tfexec.Var(fmt.Sprintf("topics=%s", jsonTopics))
-	// Ensure terraform destroy runs once per TF path.
-	RegisterDestroy(cfg.TFPath, func() {
+
+	logger().Infof("provisioning Kafka cluster %q in namespace %q", kafkaClusterName, kafkaNamespace)
+	RegisterDestroy("strimzi", func() {
 		logger().Info("destroying provisioned Kafka infrastructure...")
-		if err := tf.Destroy(context.Background(), topicsVar, namespaceVar, nameVar); err != nil {
-			logger().Error(err)
+		if err := execCommand(context.Background(),
+			"kubectl", "delete", "--ignore-not-found", "namespace", kafkaNamespace,
+		); err != nil {
+			logger().Errorf("error deleting Kafka namespace %q: %w", kafkaNamespace, err)
 		}
 	})
-	if err := tf.Apply(ctx, topicsVar, namespaceVar, nameVar); err != nil {
-		return fmt.Errorf("failed to run terraform apply: %w", err)
-	}
-	tfOutput, err := tf.Output(ctx)
-	if err != nil {
-		return fmt.Errorf("failed to run terraform output: %w", err)
-	}
-	if raw, ok := tfOutput[KafkaBrokersKey]; ok {
-		var brokers []string
-		if err := json.Unmarshal(raw.Value, &brokers); err != nil {
-			return fmt.Errorf("failed to unmarshal brokers: %w", err)
-		}
-		if len(brokers) > 0 {
-			SetKafkaBrokers(brokers...)
-		}
-	}
-	dt := string(bytes.TrimSpace(tfOutput[KafkaDeploymentTypeKey].Value))
-	// If PortForward is set to true, and the deployment_type is k8s, start
-	// forwarding traffic to/from kafka to localhost.
-	if cfg.PortForward && dt == `"k8s"` {
-		var err error
-		runOnce.Do(func() {
-			// NOTE(marclop) These assume terraform uses the strimzi operator.
-			if cfg.PortForwardResource == "" {
-				cfg.PortForwardResource = fmt.Sprintf("%s-kafka-bootstrap", cfg.Name)
-			}
-			if cfg.PortForwardMapping == "" {
-				cfg.PortForwardMapping = "9093:9093"
-			}
-			err = portforwardKafka(
-				ctx,
-				cfg.Namespace,
-				cfg.Name,
-				cfg.PortForwardResource,
-				cfg.PortForwardMapping,
-			)
-		})
-		if err != nil {
-			return fmt.Errorf("failed to run port forward: %w", err)
-		}
+
+	// Install Strimzi using Helm.
```

Comment:
There are some complications in installing the strimzi helm chart multiple times in the same cluster when multiple PRs cause concurrent installations of strimzi.

There is a conflict with ClusterRoles and ClusterRole bindings.

Reply:
In https://github.com/elastic/apm-queue/pull/170 I avoided this, by removing the strimzi part and installing a permantent strimzi operator that listens on any namespace.

Reply:
Thanks, I've remove the Strimzi installation from this PR. Longer term I think we should probably stop relying on Strimzi, and try to use a single Kafka broker pod with ephemeral storage.